src/coreclr/vm/customattribute.cpp

```diff
@@ -1118,7 +1118,7 @@ BOOL COMCustomAttribute::CopyArrayVAL(BASEARRAYREF pArray, int nElements, BYTE *
     int sizeData;   // = size * 2; with integer overflow check
     if (!ClrSafeInt<int>::multiply(nElements, sizeof(T), sizeData))
         return FALSE;
-    if (*pBlob + sizeData < *pBlob)     // integer overflow check
+    if (sizeData > endBlob - *pBlob)     // integer overflow check
```

Comment:
I think this removes the need for the range check in the following line.

Reply:
This is the same logic as the previous comment. Checking the number itself versus checking the actual indexed memory.

Reply:
Oops. You're right here.